framework/include/userobjects/DomainUserObject.h

```diff
@@ -71,6 +71,11 @@ class DomainUserObject : public UserObject,
    */
   virtual void executeOnInternalSide() {}
 
+  /**
+   * execute method that is called during ComputeUserObjects::onExternalSide
+   */
+  virtual void executeOnExternalSide(const Elem * /*elem*/, unsigned int /*side*/) {}
```

Comment:
this should be able to follow the pattern of other methods and not take any arguments

Reply:
That would be lovely, but setting `_current_elem` and `_currrent_side` comes with a considerable overhead, and - at first glace - doesn't seem possible with the current FEProblem/Assembly APIs (which require a boundary to be defined on the side). Any suggestions?

Reply:
Well in https://github.com/idaholab/moose/pull/26477/ I'm actually removing the boundary ID argument because it's used for absolutely nothing so you could just pass 0 or an invalid boundary ID if you wish. Wouldn't this "considerable overhead" be necessary for the unwary domain user object developer who assumes they can couple in a variable/material property, etc. and get meaningful evaluaitons?

Reply:
You could possibly do some caching to check whether an external face has been reinit'd in `ComputeUserObjectsThread::onBoundary` and then potentially skip the reinit in `onExternalSide` (although that wouldn't be relevant to your use case here)

Reply:
My reasoning was that any object that would require initialized variables on a boundary should operate `onBoundary`.

Reply:
alright fine

Reply:
It looks like you just changed it to what I was suggesting? You had convinced me that your way was better haha

Reply:
Yeah, the change is now causing a segfault. I'm simply adding

```
_fe_problem.reinitElemFace(elem, side, Moose::INVALID_BOUNDARY_ID, _tid);
```

Reply:
well you can change it back

Reply:
Yeah, but I would actually like to understand _why_ it is segfaulting now...

Reply:
correct

Reply:
It's an invalid access to `FEMap::dpsidxi_map` at `src/fe/fe_xyz_map.C:75` (that vector has a size of zero here).

Reply:
Can you paste the backtrace?

Reply:
Sure, here it is:
```
Thread 1 "moose_test-dbg" received signal SIGSEGV, Segmentation fault.
0x00007ffff1252432 in std::vector<double, std::allocator<double> >::operator[] (__n=<optimized out>, this=<optimized out>) at /home/schwd/mambaforge3/envs/moose/x86_64-conda-linux-gnu/include/c++/10.4.0/bits/stl_vector.h:1046
1046		return *(this->_M_impl._M_start + __n);
(gdb) bt
#0  0x00007ffff1252432 in std::vector<double, std::allocator<double> >::operator[] (__n=<optimized out>, this=<optimized out>) at /home/schwd/mambaforge3/envs/moose/x86_64-conda-linux-gnu/include/c++/10.4.0/bits/stl_vector.h:1046
#1  libMesh::FEXYZMap::compute_face_map (this=0x5555560964c0, dim=<optimized out>, qw=..., side=0x555556c87560) at ../src/fe/fe_xyz_map.C:75
#2  0x00007ffff124f7c3 in libMesh::FEXYZ<2u>::compute_face_values (this=this@entry=0x555556096160, elem=elem@entry=0x555555f06420, side=0x555556c87560, qw=...) at ../src/fe/fe_xyz_boundary.C:162
#3  0x00007ffff12508a4 in libMesh::FEXYZ<2u>::reinit (this=0x555556096160, elem=0x555555f06420, s=<optimized out>, pts=<optimized out>, weights=<optimized out>) at ../src/fe/fe_xyz_boundary.C:121
#4  0x00007ffff4ec6652 in Assembly::reinitFEFace (this=0x555555fd48b0, elem=0x555555f06420, side=0) at /home/schwd/Programs/moose-moving-boundary/framework/src/base/Assembly.C:1270
#5  0x00007ffff4ecdc04 in Assembly::reinit (this=0x555555fd48b0, elem=0x555555f06420, side=0) at /home/schwd/Programs/moose-moving-boundary/framework/src/base/Assembly.C:1946
#6  0x00007ffff673f64f in FEProblemBase::reinitElemFace (this=0x555555f839d0, elem=0x555555f06420, side=0, bnd_id=-1, tid=0) at /home/schwd/Programs/moose-moving-boundary/framework/src/problems/FEProblemBase.C:1945
#7  0x00007ffff524109d in ComputeUserObjectsThread::onExternalSide (this=0x7fffffffa8c0, elem=0x555555f06420, side=0) at /home/schwd/Programs/moose-moving-boundary/framework/src/loops/ComputeUserObjectsThread.C:260
#8  0x00007ffff4f6b7c0 in ThreadedElementLoopBase<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> >::operator() (this=0x7fffffffa8c0, range=..., bypass_threading=false)
    at /home/schwd/Programs/moose-moving-boundary/framework/build/header_symlinks/ThreadedElementLoopBase.h:302
#9  0x00007ffff67c3b7b in libMesh::Threads::parallel_reduce<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*>, ComputeUserObjectsThread> (range=..., body=...)
    at /home/schwd/Programs/moose-moving-boundary/scripts/../libmesh/installed/include/libmesh/threads_pthread.h:370
#10 0x00007ffff6760fb5 in FEProblemBase::computeUserObjectsInternal(MooseEnumItem const&, Moose::AuxGroup const&, TheWarehouse::QueryCache<>&) (this=0x555555f839d0, type=..., group=@0x7fffffffae50: Moose::POST_AUX, primary_query=...)
    at /home/schwd/Programs/moose-moving-boundary/framework/src/problems/FEProblemBase.C:4375
#11 0x00007ffff675f2aa in FEProblemBase::computeUserObjects (this=0x555555f839d0, type=..., group=@0x7fffffffae50: Moose::POST_AUX) at /home/schwd/Programs/moose-moving-boundary/framework/src/problems/FEProblemBase.C:4280
#12 0x00007ffff675e0e2 in FEProblemBase::execute (this=0x555555f839d0, exec_type=...) at /home/schwd/Programs/moose-moving-boundary/framework/src/problems/FEProblemBase.C:4128
#13 0x00007ffff5054cae in FixedPointSolve::solveStep (this=0x555555fdf370, begin_norm=@0x555556db91d0: 0, end_norm=@0x55555673a860: 0, transformed_dofs=...)
    at /home/schwd/Programs/moose-moving-boundary/framework/src/executioners/FixedPointSolve.C:457
#14 0x00007ffff50538b6 in FixedPointSolve::solve (this=0x555555fdf370) at /home/schwd/Programs/moose-moving-boundary/framework/src/executioners/FixedPointSolve.C:278
#15 0x00007ffff505c355 in Steady::execute (this=0x555555fdee40) at /home/schwd/Programs/moose-moving-boundary/framework/src/executioners/Steady.C:83
#16 0x00007ffff4f17eac in MooseApp::executeExecutioner (this=0x5555557a8b20) at /home/schwd/Programs/moose-moving-boundary/framework/src/base/MooseApp.C:1145
#17 0x00007ffff2a316f5 in MooseTestApp::executeExecutioner (this=0x5555557a8b20) at /home/schwd/Programs/moose-moving-boundary/test/src/base/MooseTestApp.C:87
#18 0x00007ffff4f1ed14 in MooseApp::run (this=0x5555557a8b20) at /home/schwd/Programs/moose-moving-boundary/framework/src/base/MooseApp.C:1527
#19 0x0000555555557db0 in Moose::main<MooseTestApp> (argc=8, argv=0x7fffffffbf68) at /home/schwd/Programs/moose-moving-boundary/framework/build/header_symlinks/MooseMain.h:47
#20 0x00005555555575f4 in main (argc=8, argv=0x7fffffffbf68) at /home/schwd/Programs/moose-moving-boundary/test/src/main.C:16
```

Reply:
I think you should revert this change. Even when the crashes are fixed you have diffs